modules/renter/contractor/downloader.go

```diff
@@ -79,7 +80,12 @@ func (hd *hostDownloader) Sector(root crypto.Hash) ([]byte, error) {
 
 	hd.contractor.mu.Lock()
 	hd.contractor.contracts[contract.ID] = contract
-	hd.contractor.saveSync()
+	cpath := fmt.Sprintf("contracts.%s", contract.ID.String())
+	hd.contractor.persist.update(
+		cpath+".lastrevision", contract.LastRevision,
+		cpath+".lastrevisiontxn", contract.LastRevisionTxn,
+		cpath+".downloadspending", contract.DownloadSpending,
+	)
```

Comment:
hmm.... actually not great because update == fsync and we don't actually want to fsync every time we change a field.